pkg/kube/workload.go

```diff
@@ -0,0 +1,65 @@
+package kube
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+	kubeAPICoreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	kubeAPIMetaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// GenericWorkload is a base implementation with some free methods for inherited structs
+type GenericWorkload struct {
+	Name        string
+	Namespace   string
+	PodSpec     kubeAPICoreV1.PodSpec
+	ObjectMeta  kubeAPIMetaV1.ObjectMeta
+	Kind        string
+	CreatedTime time.Time
+}
+
+// NewGenericWorkload builds a new workload for a given Pod
+func NewGenericWorkload(originalResource kubeAPICoreV1.Pod, dynamicClientPointer *dynamic.Interface, restMapperPointer *meta.RESTMapper) GenericWorkload {
+	workload := GenericWorkload{}
+	workload.Name = originalResource.Name
+	workload.Namespace = originalResource.Namespace
```

Comment:
@baderbuddy I can't figure out why `originalResource.Name` is defined, [based on the godoc](https://godoc.org/k8s.io/api/core/v1#Pod)

Looks like we're tracking `Name` and `Namespace` in addition to `ObjectMeta` - any thoughts on getting rid of one or the other?

Reply:
Yeah using ObjectMeta to get at Name and Namespace makes sense. I'm pretty sure the `.Name` and `.Namespace` are just shorthand.

Reply:
Ohh I misunderstood your question. Ooh that's a great point. Right now we keep the Pod's Meta no matter what, but the Name matches the top level controller. I could see value in the metadata especially for Annotations, but for the most part I think just the name and Namespace are all we really need.